src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -254,8 +263,11 @@ export function SavedObjectProvider(Promise, Private, Notifier, confirmModalProm
       this.lastSavedTitle = this.title;
 
       return Promise.try(() => {
-        parseSearchSource(meta.searchSourceJSON);
+        parseSearchSource(meta.searchSourceJSON, resp.references);
         return this.hydrateIndexPattern();
+      }).then(() => {
+        injectReferences.call(this, resp.references);
```

Comment:
This can probably be moved after `parseSearchSource` and implemented in a way that doesn't touch `this`.